### PR TITLE
[RFC] - Release process changes for Minor / Patch release

### PR DESCRIPTION
**Pull request by libscott**
Changes:

* Minor releases have a branch cut from "master", e.g. `0.9`.
* Patch releases are applied to the minor release branch.
* `Release_Process.md` describes both processes separately.

https://github.com/bigchaindb/bigchaindb/blob/release-process-updates/Release_Process.md

### CODE REVIEW

File: Release_Process.md [hunk 1]
```diff
@@ -1,23 +1,48 @@
...
+   - Target: the release branch that was just pushed
+   - Title: Same as tag name
+   - Description: The body of the changelog entry (Added, Changed etc)
+1. Publish the release on Github
+1. Generate the release tarball with `python setup.py sdist`. Upload the release to Pypi.
 1. Login to readthedocs.org as a maintainer of the BigchainDB Server docs.
    Go to Admin --> Versions and under **Choose Active Versions**, make sure that the new version's tag is
    "Active" and "Public"
-
-After the release:
```
Comment (ttmc) [inline Release_Process.md]: Why did you remove this section?

File: Release_Process.md [hunk 2]
```diff
@@ -1,23 +1,48 @@
...
+1. Fill in the details:
+   - Tag version: version number preceeded by 'v', ie: "v0.9.1"
+   - Target: the release branch that was just pushed
+   - Title: Same as tag name
+   - Description: The body of the changelog entry (Added, Changed etc)
+1. Publish the release on Github
+1. Generate the release tarball with `python setup.py sdist`. Upload the release to Pypi.
 1. Login to readthedocs.org as a maintainer of the BigchainDB Server docs.
    Go to Admin --> Versions and under **Choose Active Versions**, make sure that the new version's tag is
    "Active" and "Public"
```
Comment (ttmc) [inline Release_Process.md]: I guess we should add some more ReadTheDocs details to the end of this:
```text
   "Active" and "Public", and make sure the new version's branch
   (without the 'v' in front) is _not_ active
1. Also in readthedocs.org, go to Admin --> Advanced Settings
   and make sure that "Default branch:" (i.e. what "latest" points to)
   is set to the new release's tag, e.g. `v0.15.3`. (Don't miss the 'v' in front.)
```

File: Release_Process.md (hunk 1 above)
Comment (libscott) [inline Release_Process.md]: I didn't, I just moved it so it's the last step in a minor version release.
